### PR TITLE
Add tests for EntryComperatorTest

### PR DESCRIPTION
**Pull request by Luggas4you**
Add testPresentFieldsWithKeyField & testPresentFieldsWithoutKeyField for code coverage as well as reducing code. 

### Mandatory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by Siedlerchr — APPROVED** [review]

File: src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java [hunk 1]
```diff
@@ -4,131 +4,128 @@
...
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 class EntryComparatorTest {
+    private BibEntry e1;
+    private BibEntry e2;
+    @BeforeEach
+    void setUp() {
+            e1 = new BibEntry();
+            e2 = new BibEntry();
+    }
```
Comment (koppor) [inline src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java:19]: I know, this was lots of effort. Please revert this change. The tests were more readable before the update.

File: src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java [hunk 2]
```diff
@@ -4,131 +4,128 @@
...
         assertEquals(-1, new EntryComparator(false, false, InternalField.KEY_FIELD).compare(e1, e2));
     }
-}
 
+    @Test
+    void testPresentFieldsWithoutKeyField() {
+        e1.withField(StandardField.AUTHOR, "Stephen King");
+        e2.withField(StandardField.AUTHOR, "");
+        assertEquals(-1, new EntryComparator(true, false, InternalField.KEY_FIELD).compare(e1, e2));
+    }
```
Comment (koppor) [inline src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java:123]: This test is OK.

File: src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java [hunk 3]
```diff
@@ -4,131 +4,128 @@
...
 
+    @Test
+    void testPresentFieldsWithKeyField() {
+        e1.withField(StandardField.AUTHOR, "Stephen King");
+        e2.withField(StandardField.AUTHOR, "");
+        assertEquals(-1, new EntryComparator(true, false, StandardField.AUTHOR).compare(e1, e2));
```
Comment (koppor) [inline src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java:129]: This test is also ok

File: src/test/java/org/jabref/logic/bibtex/comparator/FieldComparatorTest.java [hunk 4]
```diff
@@ -250,4 +250,15 @@ private static Stream<Arguments> nullTests() {
...
+    @Test
+    public void compareAuthorField() throws Exception {
+        FieldComparator comparator = new FieldComparator(StandardField.AUTHOR);
+        BibEntry bigger = new BibEntry()
+                .withField(StandardField.AUTHOR, "Freund, Lucas");
+        BibEntry smaller = new BibEntry()
+                .withField(StandardField.AUTHOR, "Gärtner, Lara");
+
+        assertEquals(1, comparator.compare(smaller, bigger));
+    }
```
Comment (koppor) [inline src/test/java/org/jabref/logic/bibtex/comparator/FieldComparatorTest.java:263]: Nice one!

**Review by koppor — CHANGES_REQUESTED** [review]
Added tests look good. However, the refactoring needs to be undone.

**Review by koppor — APPROVED** [review]

**Pull request merged**